### PR TITLE
Spi plugin port count

### LINKED ISSUE #1379: SPI Plugin: Port count maximum is 8
Why is the Port count maximum set to 8?
i think it is the `UIntValidator` in this line that constrains this:
[plugins/spi/SPIDevice.cpp#L240](https://github.com/OpenLightingProject/ola/blob/master/plugins/spi/SPIDevice.cpp#L240)
```
m_preferences->SetDefaultValue(PortCountKey(), UIntValidator(1, 8), 1);
m_preferences->SetDefaultValue(SyncPortKey(), IntValidator(-2, 8), 0);
```

i would like to use about 10 universes without 'pixel-stacking' and map them to 10 ports with the software backend.
are there any things limiting this? (besides speed..)
otherwise i would make a pull request changing this to 32 or even higher like 256 ?!

### PR DESCRIPTION
**Pull request by s-light**
fixes #1379 
(based this on the APA102 Pixel Brightness Branch... hope that is ok)

### CODE REVIEW

**Comment by peternewman** [comment]
Blocked behind #1358 .

Reviewed as https://github.com/s-light/ola/compare/APA102PixelBrightness...s-light:SPIPlugin_PortCount .

File: plugins/spi/SPIDevice.h [hunk 1]
```diff
@@ -87,6 +87,8 @@ class SPIDevice: public ola::Device {
   static const char HARDWARE_BACKEND[];
   static const char SOFTWARE_BACKEND[];
   static const uint16_t MAX_GPIO_PIN = 1023;
+  static const uint16_t MAX_PORT_COUNT = 32;
+  static const uint32_t MAX_SPI_SPEED = 32000000;
```
Comment (peternewman) [inline plugins/spi/SPIDevice.h]: Nit, can we order these to match the main file, i.e. speed first?

Comment (peternewman) [reply plugins/spi/SPIDevice.h]: Done

**Comment by peternewman** [comment]
Would you mind addressing https://github.com/OpenLightingProject/ola/pull/1358#discussion_r170419192 in here @s-light .

**Comment by s-light** [comment]
https://github.com/OpenLightingProject/ola/pull/1358#discussion_r170419192
done

File: plugins/spi/SPIOutput.cpp [hunk 2]
```diff
@@ -198,7 +198,8 @@ SPIOutput::SPIOutput(const UID &uid, SPIBackendInterface *backend,
 
   ola::rdm::SlotDataCollection::SlotDataList sd_irgb_combined;
   sd_irgb_combined.push_back(
-      ola::rdm::SlotData::PrimarySlot(ola::rdm::SD_INTENSITY, 0));
+      ola::rdm::SlotData::PrimarySlot(ola::rdm::SD_INTENSITY,
+                                      ola::rdm::DMX_MAX_SLOT_VALUE));
```
Comment (peternewman) [inline plugins/spi/SPIOutput.cpp]: The tests, but curiously only the Linux ones, not OS X, are failing on this line, see https://travis-ci.org/OpenLightingProject/ola/jobs/345754263#L3298

You want ola::DMX_MAX_SLOT_VALUE, or probably just DMX_MAX_SLOT_VALUE with no namespace.

Comment (s-light) [reply plugins/spi/SPIOutput.cpp]: sorry - did not run the compile&test (was late and iam tired and thought - hey just do it quick ;-) )
i fix it and this time first do the compiles myself before i commit ;-)

**Comment by s-light** [comment]
done - for me it compiled and testsuite was successfully

File: plugins/spi/SPIOutput.cpp (hunk 2 above)
Comment (peternewman) [reply plugins/spi/SPIOutput.cpp]: That's fine, that's exactly what the Continuous Integration is there for, we all make mistakes; I certainly do it often enough!

**Pull request merged**